wetterdienst/__citation__.py

```diff
@@ -0,0 +1,51 @@
+# """Wetterdienst - Open weather data for humans"""
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2021, earthobservations developers.
+# Distributed under the MIT License. See LICENSE for more info.
+import logging
+from pathlib import Path
+
+import tomli
+import yaml
+from cff_from_621.lib import generate_cff_contents
+from cffconvert import Citation
+from jsonschema.exceptions import ValidationError
+
+ROOT = Path(__file__).parent.parent
+PYPROJECT_TOML = Path(ROOT / "pyproject.toml")
+CFF = Path(ROOT / "CITATION.cff")
+
+
+log = logging.getLogger(__name__)
+
+
+def generate_cff_file(width: int = 79):
+    """Almost similar to
+    https://github.com/delb-xml/cff-from-621/blob/1fd6dcd1077fed0eac91fc427a1ca76bb48bebcd/cff_from_621/lib.py#L40
+    """
+    content = Path(ROOT / "pyproject.toml").read_text()
+    raw = tomli.loads(content)
+    config = {"project": raw["tool"]["poetry"]}
+    config["project"]["authors"] = [{"name": author.split("<")[0][:-1]} for author in config["project"]["authors"]]
```

Comment:
Is this the only modification to the upstream code? I am just trying to get a grip on the details, to eventually submit your improvements back to the `cff-from-621` project.

Reply:
Those two adjustments had to be made because `cff-from-621` does not support poetry type format.